test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsights/ServiceBusRequestAndDependencyCollectionTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using LogLevel = Microsoft.Extensions.Logging.LogLevel;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests.ApplicationInsights
+{
+    public class ServiceBusRequestAndDependencyCollectionTests
+    {
+        private const string _queueName = "core-test-queue1";
+        private const string _mockApplicationInsightsKey = "some_key";
+        private readonly string _endpoint;
+
+        private RandomNameResolver _resolver;
+        private readonly TestTelemetryChannel _channel = new TestTelemetryChannel();
+        private static readonly AutoResetEvent _functionWaitHandle = new AutoResetEvent(false);
+
+        public ServiceBusRequestAndDependencyCollectionTests()
+        {
+            var config = new ConfigurationBuilder()
+                .AddEnvironmentVariables()
+                .Build();
+
+            var connStrProvider = new AmbientConnectionStringProvider(config);
+            var connectionString = connStrProvider.GetConnectionString(ConnectionStringNames.ServiceBus);
+            var connStringBuilder = new ServiceBusConnectionStringBuilder(connectionString);
+            _endpoint = connStringBuilder.Endpoint;
+        }
+
+        [Fact]
+        public async Task ServiceBusDepenedenciesAndRequestAreTracked()
+        {
+            using (var host = ConfigureHost())
+            {
+                await host.StartAsync();
+                await host.GetJobHost()
+                    .CallAsync(typeof(ServiceBusRequestAndDependencyCollectionTests).GetMethod(nameof(ServiceBusOut)), new {input = "message" + DateTime.UtcNow.ToString("o")});
+
+                _functionWaitHandle.WaitOne();
+
+                //TODO : clean me
+                await Task.Delay(5000);
```

Comment:
Note -- one thing I've been doing lately to remove these delays is to instead use `TestHelpers.Await()` and wait for the telemetries to appear. It'll auto-timeout if it takes too long, and that way the test is more deterministic.